src/libs/ck-libs/ampi/ampi.C

```diff
@@ -1722,11 +1722,11 @@ bool ampiParent::getBuiltinAttribute(int keyval, void *attribute_val) noexcept {
      * the base address in Fortran.
      * Also, MPI_WIN_SIZE is an MPI_Aint. */
     if (keyval == MPI_WIN_BASE)
-      *((void**)attribute_val) = *win_base_storage;
+      *(void ***)attribute_val = win_base_storage;
     else if (keyval == MPI_WIN_SIZE)
-      *(MPI_Aint *)attribute_val = *win_size_storage;
+      *(MPI_Aint **)attribute_val = win_size_storage;
     else
-      *(int *)attribute_val = *kv_builtin_storage;
+      *(int **)attribute_val = kv_builtin_storage;
     return true;
```

Comment:
This code would be quite a bit simpler if kv_get_builtin was merged into this function...

Reply:
Done